3.analysis/scripts/5.reconstruction.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+from sklearn.metrics import mean_squared_error
+from sklearn.preprocessing import MinMaxScaler
+import pandas as pd
+import pathlib 
+import sys
+import joblib
+import torch
+import torch.nn.functional as F
+from torch.utils.data import DataLoader, TensorDataset
+
+script_directory = pathlib.Path("../utils/").resolve()
+sys.path.insert(0, str(script_directory))
+from data_loader import load_model_data
+from model_utils import extract_latent_dims
+
+
+# In[2]:
+
+
+# Define the location of the saved models and output directory for results
+model_save_dir = pathlib.Path("../4.gene-expression-signatures/saved_models")
+output_dir = pathlib.Path("results")
+output_dir.mkdir(parents=True, exist_ok=True)
+
+
+# In[3]:
+
+
+data_directory = pathlib.Path("../0.data-download/data").resolve()
+dependency_file = pathlib.Path(f"{data_directory}/CRISPRGeneEffect.parquet").resolve()
+gene_dict_file = pathlib.Path(f"{data_directory}/CRISPR_gene_dictionary.parquet").resolve()
+
+
+# In[4]:
+
+
+# Load metadata
+metadata_df_dir = pathlib.Path("../0.data-download/data/metadata_df.parquet")
+metadata = pd.read_parquet(metadata_df_dir)
+print(metadata.shape)
+
+#Load dependency data
+dependency_df, gene_dict_df = load_model_data(dependency_file, gene_dict_file)
+dependency_df.head()
+
+
+# In[5]:
+
+
+# Initialize the MinMaxScaler
+scaler = MinMaxScaler()
+
+# Apply the scaler to the numeric columns
+dependency_df[dependency_df.select_dtypes(include=['float64', 'int']).columns] = scaler.fit_transform(
+    dependency_df.select_dtypes(include=['float64', 'int'])
```

Comment:
Consider being more explicit about the data types found here. Oftentimes `int` is shorthand or translated to `int64` and there are others which could appear depending on how the data were serialized or read (e.g. `int16`, `int32`, and the same for float). You could use something like the following to expand to all numeric types.

```suggestion
dependency_df[dependency_df.select_dtypes(include="number").columns] = scaler.fit_transform(
    dependency_df.select_dtypes(include="number")
```